src/pages/PasswordRecovery/index.js

```diff
@@ -0,0 +1,247 @@
+      if (ct[1] === 'unmet') {
+        return false;
+      }
+    }
+
+    // return if any error is found
+    for (const err of Object.entries(errors)) {
+      if (err[1] !== '') {
+        return false;
+      }
+    }
+
+    // if we reach up to this point,
+    // the form is validated.
+    return true;
+  };
+
+  useEffect(
+    () => {
+      if (validateForm()) {
+        setEnableSubmitButton(true);
+      } else {
+        setEnableSubmitButton(false);
+      }
+    },
+    [password, confirmPassword]
+  );
+
+  const hideAlert = () => {
+    setShowAlert(false);
+  };
+
+  // We have certain constraints on the password.
+  // Depending upon if these are met, we either display
+  // or hide them.
+  const getSplitDisplayProperty = (name, type) => {
+    if (constraints[name] === 'met' && type === 'check') {
+      return 'block';
+    }
+    if (constraints[name] === 'unmet' && type === 'close') {
+      return 'block';
+    }
+    return 'none';
+  };
+
+  const handlePasswordChange = pwd => {
+    setPassword(pwd);
+    setConstraints({
+      ...constraints,
+      passwordLength: pwd.length >= 8 ? 'met' : 'unmet',
+      passwordSpecialChars: !/^[A-Za-z0-9 ]+$/.test(pwd) ? 'met' : 'unmet',
+      passwordContainsNumber: /\d/.test(pwd) ? 'met' : 'unmet',
+      passwordBlockLetter: /[A-Z]/.test(pwd) ? 'met' : 'unmet',
+    });
+    // edge case where user deliberately
+    // edits the password field, even when
+    // confirm password is not empty.
+    if (pwd === confirmPassword) {
+      setErrors({ ...errors, passwordConfirm: '' });
+    } else {
+      setErrors({ ...errors, passwordConfirm: 'The above passwords do not match!' });
+    }
+  };
+
+  const handleConfirmPasswordChange = pwd => {
+    setConfirmPassword(pwd);
+    if (pwd !== password) {
+      setErrors({ ...errors, confirmPassword: 'The above passwords do not match' });
+    } else {
+      setErrors({ ...errors, confirmPassword: '' });
+    }
+  };
+
+  const form = (
+    <div className={styles.section}>
+      {showAlert && (
+        <Alert
+          className={styles.alert}
+          variant="success"
+          title="Successfully changed your password!"
+          action={<AlertActionCloseButton onClose={hideAlert} />}
+        />
+      )}
+      <Title headingLevel="h4" size="xl">
+        <b>Recover password</b>
+      </Title>
+      <br />
+      <Form>
+        <FormGroup label="Password" isRequired fieldId="horizontal-form-password">
+          <TextInput isRequired type="password" onChange={handlePasswordChange} />
+          <p className={styles.error}>{errors.password}</p>
+        </FormGroup>
+        <FormGroup label="Confirm Password" isRequired fieldId="horizontal-form-confirm-password">
+          <TextInput isRequired type="password" onChange={handleConfirmPasswordChange} />
+          <p className={styles.error}>{errors.confirmPassword}</p>
+        </FormGroup>
+        <div>
+          <h4>Password must contain at least</h4>
```

Comment:
Any way we can leverage the same from the normal signup page?

Reply:
Do we create this (split components for password constraints) as a separate component and reference them in each page component that we need.

Reply:
Fixed!